### PR TITLE
AF + threat vault - auto key

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/33585

## Description
This PR will allow  >6.2.0 versions AF integrations to use the `demisto.getAutoFocusApiKey()` method to set the AF API key.
hence, the parameter is now non mandatory.
also, a config to override the default has been added, set by default to false.
Added to threat vault as well.

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests - added a config without an API key in the test_conf. 1 AF v2 TPB should run fromverison 6.2.0 without it.
- [x] Documentation

### CODE REVIEW

**Comment by bakatzir** [comment]
@kirbles19 Can you revise the added parameter name/additionalinfo?

**Comment by bakatzir** [comment]
@DeanArbel / @yuvalbenshalom Do you think the check is generic enough to go into CSP? I'm contemplating.

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml [hunk 1]
```diff
@@ -21,6 +21,11 @@ configuration:
   - F - Reliability cannot be judged
   required: true
   type: 15
+- additionalinfo: Whether to override the default AutoFocus API Key given by the Cortex XSOAR platform.
```
Comment (amshamah419) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml:24]:
```suggestion
- additionalinfo: Whether to override the default AutoFocus API key given by the Cortex XSOAR platform.
```

File: Packs/AutoFocus/Integrations/AutofocusV2/README.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ To get your API key, you need to add an authorization code, and then activate th
 Use the API key when configuring the integration.
 For more information on activating the license see [Activating AutoFocus Licenses](https://docs.paloaltonetworks.com/autofocus/autofocus-admin/get-started-with-autofocus/activate-autofocus-licenses.html).
 
-## Configure AutoFocus V2 on Demisto
+## Configure AutoFocus V2 on Cortex XSOAR
```
Comment (amshamah419) [inline Packs/AutoFocus/Integrations/AutofocusV2/README.md:26]: :corporate:

File: Packs/AutoFocus/Integrations/AutofocusV2/README.md [hunk 3]
```diff
@@ -35,9 +35,10 @@ For more information on activating the license see [Activating AutoFocus License
    | Name | A meaningful name for the integration instance. | AutoFocus V2_instance_2 |
    | API Key | Account's private token. | N/A  |
    | Source Reliability | Reliability of the source providing the intelligence data. | B - Usually reliable |
+   | Additional Malicious Verdicts  | A comma-separated list of Palo Alto Networks verdicts to consider as malicious when calculating the DBot score.  | malware,phishing,c2 |   
+   | Override default credentials | Whether to override the default AutoFocus API Key given by the Cortex XSOAR platform. | False |
```
Comment (amshamah419) [inline Packs/AutoFocus/Integrations/AutofocusV2/README.md:39]:
```suggestion
   | Override default credentials | Whether to override the default AutoFocus API key given by the Cortex XSOAR platform. | False |   
```

File: Packs/AutoFocus/Integrations/AutofocusV2/README.md [hunk 4]
```diff
@@ -35,9 +35,10 @@ For more information on activating the license see [Activating AutoFocus License
...
    | API Key | Account's private token. | N/A  |
    | Source Reliability | Reliability of the source providing the intelligence data. | B - Usually reliable |
+   | Additional Malicious Verdicts  | A comma-separated list of Palo Alto Networks verdicts to consider as malicious when calculating the DBot score.  | malware,phishing,c2 |   
+   | Override default credentials | Whether to override the default AutoFocus API Key given by the Cortex XSOAR platform. | False |   
    | Trust any certificate (not secure) | When selected, certificates are not checked. | N/A |
-   | Use System Proxy Settings | Runs the integration instance using the proxy server (HTTP or HTTPS) that you defined in the server configuration. |  https:/<span></span>/www.markdownguide.org |
-   | Additional Malicious Verdicts  | A comma-separated list of Palo Alto Networks verdicts to consider as malicious when calculating the DBot score.  | malware,phishing,c2 |
+   | Use System Proxy Settings | Runs the integration instance using the proxy server (HTTP or HTTPS) that you defined in the server configuration. |  https:/<span></span>/www.markdownguide.org |
```
Comment (amshamah419) [inline Packs/AutoFocus/Integrations/AutofocusV2/README.md:41]: What's the need for the span tag?

File: Packs/AutoFocus/ReleaseNotes/1_1_17.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+
+#### Integrations
+##### AutoFocus Daily Feed
+- Added support for Cortex XSOAR Auto Focus API Key usage.
```
Comment (amshamah419) [inline Packs/AutoFocus/ReleaseNotes/1_1_17.md:4]:
```suggestion
- Added support to enable the usage of the Cortex XSOAR Auto Focus API key.
```

File: Packs/AutoFocus/ReleaseNotes/1_1_17.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+- Updated the Docker image to: *demisto/python3:3.9.4.18682*.
+##### Palo Alto Networks AutoFocus v2
+- Added support for Cortex XSOAR Auto Focus API Key usage.
```
Comment (amshamah419) [inline Packs/AutoFocus/ReleaseNotes/1_1_17.md:7]:
```suggestion
- Added support to enable the usage of the Cortex XSOAR Auto Focus API key.
```

File: Packs/AutoFocus/ReleaseNotes/1_1_17.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+- Updated the Docker image to: *demisto/python3:3.9.4.18682*.
+##### AutoFocus Feed
+- Added support for Cortex XSOAR Auto Focus API Key usage.
```
Comment (amshamah419) [inline Packs/AutoFocus/ReleaseNotes/1_1_17.md:10]:
```suggestion
- Added support to enable the usage of the Cortex XSOAR Auto Focus API key.
```

File: Packs/Base/ReleaseNotes/1_9_0.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPython
+- Added support for getting the AutoFocus API Key(server version 6.2.0+).
```
Comment (amshamah419) [inline Packs/Base/ReleaseNotes/1_9_0.md:4]:
```suggestion
- Added support to enable usage of the AutoFocus API key (XSOAR server version 6.2.0+).
```

File: Packs/PaloAltoNetworks_Threat_Vault/ReleaseNotes/1_0_3.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Palo Alto Networks Threat Vault
+- Added support for Cortex XSOAR Auto Focus API Key usage.
```
Comment (amshamah419) [inline Packs/PaloAltoNetworks_Threat_Vault/ReleaseNotes/1_0_3.md:4]:
```suggestion
- Added support to enable the usage of the Cortex XSOAR Auto Focus API key.
```

File: Packs/AutoFocus/Integrations/AutofocusV2/README.md (hunk 4 above)
Comment (bakatzir) [reply Packs/AutoFocus/Integrations/AutofocusV2/README.md:41]: no need, rming.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 10]
```diff
@@ -6507,3 +6507,27 @@ def search_indicators_by_version(self, from_date=None, query='', size=100, to_da
...
+    :param override_default_credentials: Whether to override the default credentials and use the
+     Cortex XSOAR given AutoFocus API Key
+    :return: No data returned
+    :rtype: ``None``
+    """
+    def __init__(self, api_key, override_default_credentials):
+        # demisto.getAutoFocusApiKey() is available from version 6.2.0
+        if not api_key:
+            if not is_demisto_version_ge("6.2.0"):  # AF API key is available from version 6.2.0
+                raise Exception('For versions earlier than 6.2.0, configure an API Key.')
```
Comment (jochman) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:6526]: It's a bad practice to use `Exception`, use `DemistoException` or create a new exception like `CredentialsException`

File: Packs/VirusTotal/ReleaseNotes/2_0_4.md [hunk 11]
```diff
@@ -1,4 +1,4 @@
 
 #### Integrations
```
Comment (jochman) [inline Packs/VirusTotal/ReleaseNotes/2_0_4.md:2]: revert

**Review by jochman — COMMENTED** [review]
Just change the raw Exception to specific one.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**